internal/db/p2p.go

```diff
@@ -0,0 +1,223 @@
+// unsubscribes from their topics. It will error if the provided
+// collection names are invalid.
+func (db *DB) RemoveP2PCollections(ctx context.Context, collectionNames ...string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, false)
+	if err != nil {
+		return err
+	}
+	defer txn.Discard(ctx)
+
+	err = db.p2p.RemoveP2PCollections(ctx, collectionNames...)
+	if err != nil {
+		return err
+	}
+
+	return txn.Commit(ctx)
+}
+
+// GetAllP2PCollections returns the list of persisted collection names that
+// the P2P system subscribes to.
+func (db *DB) GetAllP2PCollections(ctx context.Context) ([]string, error) {
+	if db.p2p == nil {
+		return nil, errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, true)
+	if err != nil {
+		return nil, err
+	}
+	defer txn.Discard(ctx)
+
+	return db.p2p.GetAllP2PCollections(ctx)
+}
+
+// AddP2PDocuments adds the given docIDs to the P2P system and
+// subscribes to their topics. It will error if any of the provided
+// docIDs are invalid.
+func (db *DB) AddP2PDocuments(ctx context.Context, docIDs ...string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, false)
+	if err != nil {
+		return err
+	}
+	defer txn.Discard(ctx)
+
+	err = db.p2p.AddP2PDocuments(ctx, docIDs...)
+	if err != nil {
+		return err
+	}
+
+	return txn.Commit(ctx)
+}
+
+// RemoveP2PDocuments removes the given docIDs from the P2P system and
+// unsubscribes from their topics. It will error if the provided
+// docIDs are invalid.
+func (db *DB) RemoveP2PDocuments(ctx context.Context, docIDs ...string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, false)
+	if err != nil {
+		return err
+	}
+	defer txn.Discard(ctx)
+
+	err = db.p2p.RemoveP2PDocuments(ctx, docIDs...)
+	if err != nil {
+		return err
+	}
+
+	return txn.Commit(ctx)
+}
+
+// GetAllP2PDocuments returns the list of persisted docIDs that
+// the P2P system subscribes to.
+func (db *DB) GetAllP2PDocuments(ctx context.Context) ([]string, error) {
+	if db.p2p == nil {
+		return nil, errors.New("no p2p system comfigured")
+	}
+	ctx, txn, err := ensureContextTxn(ctx, db, true)
+	if err != nil {
+		return nil, err
+	}
+	defer txn.Discard(ctx)
+
+	return db.p2p.GetAllP2PDocuments(ctx)
+}
+
+// SyncDocuments requests the latest versions of specified documents from the network
+// and synchronizes their DAGs locally. It doesn't automatically subscribe
+// to the documents or their collection for future updates.
+// context.WithTimeout can be used to set a timeout for the operation.
+//
+// WARNING: This function does not respect transactions.
+func (db *DB) SyncDocuments(ctx context.Context, collectionName string, docIDs []string) error {
+	if db.p2p == nil {
+		return errors.New("no p2p system comfigured")
```

Comment:
todo: Move to `errors.go` file.

todo: typo:

```suggestion
		return errors.New("no p2p system configured")
```

Reply:
done